### PR TITLE
refactor(email): migrate arg extraction onto internal/tools/toolargs (#965)

### LINKED ISSUE #965: Tool-arg extraction helpers duplicated across packages; finish migrating onto internal/tools/toolargs
Status updated 2026-06-25.

The canonical shared package landed in PR 1001, Refs 965, merged 2026-05-28, commit e96cc6c, as internal/tools/toolargs, not the originally-sketched internal/tools/args. The rename was deliberate and documented: handlers conventionally name their argument map args, so a package named args would be shadowed at almost every call site. toolargs.String reads cleanly with no alias. That PR delivered the package plus tests and migrated one of the three duplicating packages. Two parts of the original migration plan remain.

Done: Canonical package internal/tools/toolargs with full translator API String TrimmedString Int IntOr IntOK Bool BoolOr BoolOK HasBool StringSlice Uint32Slice and table-driven tests covering the JSON-coercion edge cases, including float64 integrality, numeric strings, string bools, the 930 omitted-bool default, and uint32 range filtering. Also migrated internal/tools loop and model-registry tools by deleting the mr and ld string and int prefixed helpers and repointing about 71 call sites. Loop-specific decoders decodeLoopSpecArg and decodeLoopLaunchArg and registry-specific helpers correctly stay put.

Remaining task one: Migrate internal/channels/email onto toolargs. It still carries the full prefixed set at internal/channels/email/tools.go lines 500-622 named stringArg intArg boolArg boolArgDefault uint32SliceArg coerceUint32 stringSliceArg with its own duplicated coercion and tests in tools_test.go, and does not import toolargs. This is the highest-value piece: email is the package that still forks the coercion and bool-default contract this issue was filed about, the 930 bool-default lives only here. BoolOr replaces boolArgDefault; Uint32Slice and StringSlice already exist in toolargs, so the migration is mostly call-site repointing plus folding or retiring the email-specific tests. Verify the email warn-logging behavior is preserved or intentionally dropped.

Remaining task two: Inline assertion sweep. Roughly 255 non-test value-type assertion sites remain across internal, the issue originally cited 167. Replace with toolargs getters opportunistically as tools are touched, or in a focused pass.

Original smell for context: Three tool packages each defined their own prefixed copy of the same JSON-arg extraction helpers stringArg ldStringArg mrStringArg, prefixed only because they would collide. The prefixes were a workaround for a missing shared home; subtle coercion differences risked drift. toolargs is now that home. Lineage: surfaced in a wrong-fix shape by the v0.9.3 code-path audit on PR 963; canonical package and first migration landed in PR 1001 commit e96cc6c.

### PR DESCRIPTION
**Pull request by thane-developer**
Refs #965 (the "remaining task one" — migrate `internal/channels/email` onto the canonical `toolargs`).

## Background

The canonical `internal/tools/toolargs` package landed in #1001 and the loop + model-registry tools migrated onto it. `internal/channels/email` was the **last package still forking** the same JSON-arg extraction helpers — `stringArg / intArg / boolArg / boolArgDefault / uint32SliceArg / coerceUint32 / stringSliceArg` — which is the exact cross-package duplication this issue was filed about, and where the #930 bool-default contract lived.

## Change

- Repoint all **36 call sites** onto `toolargs` (`String / Int / Bool / BoolOr / StringSlice / Uint32Slice`).
- Delete the forked helpers and their now-redundant unit tests (toolargs already has table-driven coverage of the coercion edge cases, incl. the #930 omitted-bool default and uint32 range filtering).
- Drop the now-unused `strconv` import.

Net **−437 / +38**. With email migrated, all three formerly-forking packages now consume `toolargs` — the duplication smell is gone.

## Behavior notes (per the issue's "verify warn-logging" ask)

Two warn logs are **intentionally dropped** to match `toolargs`' silent contract (and the prior loop/model-registry migrations): `"failed to parse int argument"` and `"skipped invalid UID element"`. `toolargs` is also **stricter / more robust** — it rejects fractional floats (the old `intArg` truncated) and additionally coerces `int32/int64/json.Number` and string-encoded bools. The affected args are counts/UIDs where reject-to-0 is safe.

The **#930 regression guard** (`TestParseMarkAction`: omitted `add` → `true`) is retained and passes against `toolargs.BoolOr`.

## Test plan

- [x] `just ci` green (email pkg tests, race; full gate)
- [x] `TestParseMarkAction` (#930 guard) passes against `toolargs.BoolOr`
- [x] No residual forked-helper references; `large_files` metric dropped 48→47
- [x] gofmt clean; non-test build + race tests pass

## Not in this PR

**Task two** of #965 — the ~255 inline value-type assertion sites across `internal/` — is opportunistic/large per the audit and is best done as tools are touched or in a focused sweep. Proposing to track it separately (and re-scope/close #965 once this lands); see PR discussion.

### CODE REVIEW

File: internal/channels/email/tools.go [hunk 1]
```diff
@@ -59,9 +59,9 @@ func (t *Tools) HandleList(ctx context.Context, args map[string]any) (string, er
 
 // HandleRead reads a single email by UID.
 func (t *Tools) HandleRead(ctx context.Context, args map[string]any) (string, error) {
-	uid := uint32(intArg(args, "uid"))
-	folder := stringArg(args, "folder")
-	account := stringArg(args, "account")
+	uid := uint32(toolargs.Int(args, "uid"))
```
[the remark this answers is not in the document]
Comment (thane-developer) [reply internal/channels/email/tools.go]: Fixed in 11ca3ba. Added a singular `toolargs.Uint32`/`Uint32OK` getter (the single-value form of `Uint32Slice`'s coercion — rejects nil/negative/fractional/<1/out-of-range) and repointed all four singular-uid sites onto it, so a negative or out-of-range UID now reads as "not provided" (0) instead of wrapping to a large valid-looking UID. New table-driven tests cover the negative-no-wrap and ceiling cases. (This was pre-existing behavior carried over from the old `intArg`; good catch.)

File: internal/channels/email/tools.go [hunk 2]
```diff
@@ -179,14 +179,14 @@ func (t *Tools) HandleMark(ctx context.Context, args map[string]any) (string, er
...
+		Flag:    toolargs.String(args, "flag"),
+		Add:     toolargs.BoolOr(args, "add", true),
+		Account: toolargs.String(args, "account"),
 	}
-	action.UIDs = uint32SliceArg(args, "uids")
+	action.UIDs = toolargs.Uint32Slice(args, "uids")
 	if len(action.UIDs) == 0 {
-		if uid := uint32(intArg(args, "uid")); uid != 0 {
+		if uid := uint32(toolargs.Int(args, "uid")); uid != 0 {
 			action.UIDs = []uint32{uid}
```
[the remark this answers is not in the document]
Comment (thane-developer) [reply internal/channels/email/tools.go]: Fixed in 11ca3ba. Added a singular `toolargs.Uint32`/`Uint32OK` getter (the single-value form of `Uint32Slice`'s coercion — rejects nil/negative/fractional/<1/out-of-range) and repointed all four singular-uid sites onto it, so a negative or out-of-range UID now reads as "not provided" (0) instead of wrapping to a large valid-looking UID. New table-driven tests cover the negative-no-wrap and ceiling cases. (This was pre-existing behavior carried over from the old `intArg`; good catch.)

File: internal/channels/email/tools.go [hunk 3]
```diff
@@ -219,11 +219,11 @@ func (t *Tools) HandleSend(ctx context.Context, args map[string]any) (string, er
 // HandleReply replies to an existing message with threading headers.
 func (t *Tools) HandleReply(ctx context.Context, args map[string]any) (string, error) {
 	opts := ReplyOptions{
-		UID:      uint32(intArg(args, "uid")),
-		Folder:   stringArg(args, "folder"),
-		Body:     stringArg(args, "body"),
-		ReplyAll: boolArg(args, "reply_all"),
-		Account:  stringArg(args, "account"),
+		UID:      uint32(toolargs.Int(args, "uid")),
```
[the remark this answers is not in the document]
Comment (thane-developer) [reply internal/channels/email/tools.go]: Fixed in 11ca3ba. Added a singular `toolargs.Uint32`/`Uint32OK` getter (the single-value form of `Uint32Slice`'s coercion — rejects nil/negative/fractional/<1/out-of-range) and repointed all four singular-uid sites onto it, so a negative or out-of-range UID now reads as "not provided" (0) instead of wrapping to a large valid-looking UID. New table-driven tests cover the negative-no-wrap and ceiling cases. (This was pre-existing behavior carried over from the old `intArg`; good catch.)

File: internal/channels/email/tools.go [hunk 4]
```diff
@@ -308,9 +308,9 @@ func (t *Tools) HandleMove(ctx context.Context, args map[string]any) (string, er
 	}
 
 	// Parse UIDs — accept "uids" array or singular "uid".
-	opts.UIDs = uint32SliceArg(args, "uids")
+	opts.UIDs = toolargs.Uint32Slice(args, "uids")
 	if len(opts.UIDs) == 0 {
-		if uid := uint32(intArg(args, "uid")); uid != 0 {
+		if uid := uint32(toolargs.Int(args, "uid")); uid != 0 {
 			opts.UIDs = []uint32{uid}
 		}
```
[the remark this answers is not in the document]
Comment (thane-developer) [reply internal/channels/email/tools.go]: Fixed in 11ca3ba. Added a singular `toolargs.Uint32`/`Uint32OK` getter (the single-value form of `Uint32Slice`'s coercion — rejects nil/negative/fractional/<1/out-of-range) and repointed all four singular-uid sites onto it, so a negative or out-of-range UID now reads as "not provided" (0) instead of wrapping to a large valid-looking UID. New table-driven tests cover the negative-no-wrap and ceiling cases. (This was pre-existing behavior carried over from the old `intArg`; good catch.)

**Review by nugget — APPROVED** [review]
